addons/website_livechat/static/src/js/website_livechat.js

```diff
@@ -0,0 +1,34 @@
+odoo.define('website_livechat.history', function (require) {
+    var local_storage = require('web.local_storage');
+    var session = require('web.session');
+    var bus = require('bus.bus').bus;
+    var website = require('website.website');
+    var im_livechat = require('im_livechat.im_livechat');
+
+    var page = window.location.href.replace(/^.*\/\/[^\/]+/, ''),
+        page_history = local_storage.getItem('visited_page_history');
+    var url_history = JSON.parse(page_history) || [];
+    if(!_.contains(url_history, page)){
+        url_history.push(page);
+        local_storage.setItem('visited_page_history', JSON.stringify(url_history));
+    }
+    
+    im_livechat.LivechatButton.include({
+        add_message: function(data, option){
+            var self = this;
+            if (data._type === "history_command"){
+                page_history = local_storage.getItem('visited_page_history');
+                session.rpc("/livechat/history_pages", {
+                    pid: self.channel.operator_pid[0],
+                    channel_id: self.channel.id,
+                    page_history: JSON.parse(page_history),
+                }).then(function(){
+                    local_storage.removeItem('visited_page_history');
```

Comment:
pros: remove localstorage is cool. 
cons: But if I re execute the history command, I will get nothing.

I'm wondering if it's a good idea to store it in the localstorage then. Also, if a person come to visit my website, and restart a livechat session, I will get the history of its old visit.
@tde-banana-odoo wouldn't it be even better to store history in a cookie, and per livechat conversation ? (like, if I track pages only if a livechat is started).